paparazzi-preview-processor/src/main/java/app/cash/paparazzi/preview/processor/PreviewProcessor.kt

```diff
@@ -15,5 +19,47 @@ public class PreviewProcessorProvider : SymbolProcessorProvider {
 public class PreviewProcessor(
   private val environment: SymbolProcessorEnvironment
 ) : SymbolProcessor {
-  override fun process(resolver: Resolver): List<KSAnnotated> = emptyList()
+
+  private var invoked = false
+
+  override fun process(resolver: Resolver): List<KSAnnotated> {
+    if (invoked) return emptyList()
+    invoked = true
```

Comment:
what does this guard accomplish?  is there some weird recursive bug happening here?

Reply:
Ya basically the first run is from the initial source. Once we generate the new `PaparazziPreviews.kt` file with kotlin poet, it reruns the processor because the new file needs validation. Updated with more strict logic for this case instead of a generic bool.